optimade/validator/validator.py

```diff
@@ -1327,6 +1327,14 @@ def _get_available_endpoints(
                 raise ResponseError(
                     f'Illegal entry "{non_entry_endpoint}" was found in entry_types_by_format"'
                 )
+
+        # Filter out custom extension endpoints that are not covered in the specification
+        available_json_entry_endpoints = [
+            endp
+            for endp in available_json_entry_endpoints
+            if not endp.startswith("extensions/")
```

Comment:
If there are custom endpoints not under `extensions/`, then they are still not accounted for.
I think we should instead turn this upside down and only test a specific set of endpoints and leave everything else alone? Or do we have requirements for custom endpoints in the specification? Stuff like what they _have_ to return and such?

Reply:
The only constraint is that they have to be served from `extensions/<type>`, so anything that does not start with `extensions/` is fair game (this obviously relates to https://github.com/Materials-Consortia/optimade-python-tools/issues/793#issuecomment-834411926).

I think your suggestion makes sense to only validate `{"structures", "references", "calculations"}` (where `"/calculations"` will only be checked as a valid entry listing endpoint).

Reply:
I believe we at some point agreed in the specification that providers can create custom endpoints both at the root as well as under `extensions/`, hence my idea to instead focus on the entry endpoints we know about, i.e., the ones you list here.

Reply:
Yup, imagine this is something that will be discussed again this year; let's just fix this as suggested for now.